core/src/it/java/software/amazon/awssdk/http/ConnectionPoolMaxConnectionsIntegrationTest.java

```diff
@@ -53,26 +54,32 @@ public void leasing_a_new_connection_fails_with_connection_pool_timeout() throws
 
         String localhostEndpoint = "http://localhost:" + server.getPort();
 
-        AmazonHttpClient httpClient =
-                builderWithDefaultClient()
-                                .sdkHttpClient(ApacheSdkHttpClientFactory.builder()
-                                                                         .connectionTimeout(Duration.ofMillis(100))
-                                                                         .maxConnections(1)
-                                                                         .build()
-                                                                         .createHttpClient())
-                                .clientConfiguration(new LegacyClientConfiguration()
-                                                             .withMaxErrorRetry(0))
-                                .build();
+        AmazonHttpClient httpClient = HttpTestUtils.testClientBuilder()
+                                                   .clientExecutionTimeout(null)
```

Comment:
By default, the test client has a client execution timeout configured. This one specifically needs it to be disabled to prevent the test from failing intermittently.